### PR TITLE
Sync v2-5-stable with v2-5-test to release 2.5.1 

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.5.1rc1`!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,7 +21,114 @@
...
+- Make invalid characters exception more readable (#28181)
+- Bump decode-uri-component from 0.2.0 to 0.2.2 in ``/airflow/www`` (#28080)
+- Use asserts instead of exceptions for executor not started (#28019)
+- Simplify dataset ``subgraph`` logic (#27987)
+- Order TIs by map_index (#27904)
+- Additional info about Segmentation Fault in LocalTaskJob (#27381)
+
+Doc Only Changes
+^^^^^^^^^^^^^^^^
+- Update scheduler docs about low priority tasks (#28831)
```
Comment (pierrejeambrun) [inline RELEASE_NOTES.rst:123]: https://github.com/apache/airflow/pull/27235 is cherry picked but there is not entry in the change log for it.

Was it  removed purposely ?

**Review by pierrejeambrun — APPROVED** [review]
Beside my question above, LGTM.

As a sanity check I have built the image locally and started breeze, ran over 30 dags, everything went well.

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (potiuk) [reply RELEASE_NOTES.rst:123]: Yes I think it should be added.

Comment (eladkal) [reply RELEASE_NOTES.rst:123]: yes should be here as well

**Review by eladkal — APPROVED** [review]

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (ephraimbuddy) [reply RELEASE_NOTES.rst:123]: Thanks for spotting this. I guess there would be more like it. I will run the changelog command once more

**Review by potiuk — APPROVED** [review]

**Pull request merged**